### PR TITLE
feat(plugin-mocks): Added mocks for getAllPlugins query

### PR DESCRIPTION
**Pull request by jatinriverlane**
## Changes:
* add mocks for getAllPlugins query

### CODE REVIEW

**Review by samiralavi — APPROVED** [review]

File: aqueductcore/frontend/src/API/graphql/queries/getAllExtentions.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { gql, QueryHookOptions, useQuery } from "@apollo/client";
+import { GET_ALL_EXTENSIONS_TYPE } from "types/globalTypes";
+
+export const GET_ALL_EXTENTIONS = gql`
+  query getAllPlugins {
```
Comment (str-anger) [inline aqueductcore/frontend/src/API/graphql/queries/getAllExtentions.tsx:5]:
```suggestion
  query getAllExtensions {
```

I hope this well be merged soon: https://github.com/AqueductHub/aqueductcore/blob/a356e0c8f5ea11eef572cd88ee8d5a245424a4df/aqueductcore/backend/routers/graphql/query_schema.py#L107

File: aqueductcore/frontend/src/API/graphql/queries/getAllExtentions.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    plugins {
+      authors
+      description
+      functions {
+        description
+        experimentVariableName
+        name
+        parameters {
```
Comment (str-anger) [inline aqueductcore/frontend/src/API/graphql/queries/getAllExtentions.tsx:13]:
```suggestion
    extensions {
      authors
      description
      actions {
        description
        experimentVariableName
        name
        parameters {
```

File: aqueductcore/frontend/src/__mocks__/queries/getAllExtentionsMock.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+const request = {
+  query: GET_ALL_EXTENTIONS,
+};
+
+export const getAllExtension_mock = {
+  success: {
+    request,
+    result: {
+        "data": {
+          "plugins": [
```
Comment (str-anger) [inline aqueductcore/frontend/src/__mocks__/queries/getAllExtentionsMock.tsx:12]:
```suggestion
          "extensions": [
```

File: aqueductcore/frontend/src/__mocks__/queries/getAllExtentionsMock.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            {
+              "authors": "test@email.com",
+              "description": "Description of plugin.\n",
+              "functions": [
```
Comment (str-anger) [inline aqueductcore/frontend/src/__mocks__/queries/getAllExtentionsMock.tsx:16]:
```suggestion
              "actions": [
```

File: aqueductcore/frontend/src/__mocks__/queries/getAllExtentionsMock.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                      "options": null
+                    }
+                  ]
+                }
+              ]
+            },
+            {
+              "authors": "test@email.com",
+              "description": "Description of plugin.\n",
+              "functions": [
```
Comment (str-anger) [inline aqueductcore/frontend/src/__mocks__/queries/getAllExtentionsMock.tsx:68]:
```suggestion
              "actions": [
```

File: aqueductcore/frontend/src/types/globalTypes.ts [hunk 6]
```diff
@@ -1,8 +1,12 @@
 import {
+  MutationExecutePluginArgs,
   ExperimentFiltersInput,
+  PluginExecutionResult,
+  PluginFunctionInfo,
   ExperimentData,
   ExperimentFile,
   Experiments,
+  PluginInfo,
```
Comment (str-anger) [inline aqueductcore/frontend/src/types/globalTypes.ts:9]:
```suggestion
  ExtensionExecutionResult,
  ExtensionActionInfo,
  ExperimentData,
  ExperimentFile,
  Experiments,
  ExtensionInfo,
```

File: aqueductcore/frontend/src/types/globalTypes.ts [hunk 7]
```diff
@@ -1,8 +1,12 @@
 import {
+  MutationExecutePluginArgs,
```
Comment (str-anger) [inline aqueductcore/frontend/src/types/globalTypes.ts:2]:
```suggestion
  MutationExecuteExtensionArgs,
```

File: aqueductcore/frontend/src/types/globalTypes.ts [hunk 8]
```diff
@@ -41,6 +45,14 @@ export type GET_ALL_TAGS_TYPE = {
   };
 };
 
+export type GET_ALL_EXTENSIONS_TYPE = {
```
Comment (str-anger) [inline aqueductcore/frontend/src/types/globalTypes.ts:48]: here as well. Let us stick to extensions and actions.

File: aqueductcore/frontend/src/types/globalTypes.ts [hunk 9]
```diff
@@ -165,3 +182,20 @@ export interface ExperimentRecordsColumnsType {
   ellipsis?: boolean;
   format?: (value?: unknown) => string | React.ReactNode;
 }
+
+//############### Plugin types ###############//
+
+export type PluginFunctionInfoType = {
```
Comment (str-anger) [inline aqueductcore/frontend/src/types/globalTypes.ts:188]: extensions, actions

File: aqueductcore/frontend/src/types/globalTypes.ts [hunk 10]
```diff
@@ -165,3 +182,20 @@ export interface ExperimentRecordsColumnsType {
   ellipsis?: boolean;
   format?: (value?: unknown) => string | React.ReactNode;
 }
+
+//############### Plugin types ###############//
```
Comment (str-anger) [inline aqueductcore/frontend/src/types/globalTypes.ts:186]:
```suggestion
//############### Extension types ###############//
```

**Review by str-anger — CHANGES_REQUESTED** [review]
I feel like my skills don't allow me to judge the code, but for the naming, please in the commented places and everywhere else try to use `(extensions, actions)` convention.

**Comment by jatinriverlane** [comment]
This has been implemented in a different PR

**Pull request closed without merging**